app/containers/ChangeEmail/ChangeEmailForm.jsx

```diff
@@ -0,0 +1,133 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import { Field, reduxForm } from 'redux-form/immutable';
+import { FormattedMessage } from 'react-intl';
+import { translationMessages } from 'i18n';
+
+import commonMessages from 'common-messages';
+
+import H4 from 'components/H4';
+import TextInputField from 'components/FormFields/TextInputField';
+import Button from 'components/Button/Contained/InfoLarge';
+import signUpMessages from 'containers/SignUp/messages';
+import { validatePassword } from 'containers/SignUp/IHaveEOSAccountForm';
+
+import {
+  strLength3x20,
+  required,
+  validateEmail,
+} from 'components/FormFields/validate';
+
+import {
+  NEW_EMAIL_FIELD,
+  CONFIRM_EMAIL_FIELD,
+  PASSWORD_FIELD,
+  CHANGE_EMAIL_FORM,
+} from './constants';
+
+import changeEmailMessages from './messages';
+
+const ChangeEmailForm = ({
+  handleSubmit,
+  locale,
+  changeEmail,
+  changeEmailProcessing,
+}) => (
+  <div>
+    <H4 className="text-center pb-3">
+      <FormattedMessage {...commonMessages.change} />{' '}
+      <FormattedMessage {...signUpMessages.email} />
+    </H4>
+
+    <form onSubmit={handleSubmit(changeEmail)}>
+      <Field
+        name={NEW_EMAIL_FIELD}
+        disabled={changeEmailProcessing}
+        label={translationMessages[locale][changeEmailMessages.newEmail.id]}
+        component={TextInputField}
+        validate={[validateEmail, strLength3x20, required]}
```

Comment:
`strLength3x20` should not be needed here. `validateEmail` has to check the length as well for simplicity.

Reply:
fixed